### PR TITLE
Feature improve base object and component

### PR DESCRIPTION
**Pull request by vkhramtsov**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes/no

Call class methods where possible. Improve code coverage

### CODE REVIEW

**Comment by samdark** [comment]
As far as I remember, it was made like that intentionally in order to keep performance at acceptable levels.

**Comment by vkhramtsov** [comment]
Waiting for your decision. Currently you are using php 7.1, so it works faster than before

**Comment by vkhramtsov** [comment]
@samdark Any news here? Also I would like to mention that first build fails because of other tests, not mine ;)

**Comment by samdark** [comment]
The tests are OK. Cannot decide on it without performance measurements.

File: tests/framework/base/BaseObjectTest.php [hunk 1]
```diff
@@ -64,25 +65,31 @@ public function testCanSetProperty()
         $this->assertFalse($this->object->canSetProperty('Content'));
     }
 
+    /**
+     * @expectedException \yii\base\UnknownPropertyException
+     */
```
Comment (rob006) [inline tests/framework/base/BaseObjectTest.php]: Do not use annotations for expected exception: https://thephp.cc/news/2016/02/questioning-phpunit-best-practices

Comment (vkhramtsov) [reply tests/framework/base/BaseObjectTest.php]: OK, fixed. I suggest to add this to contribution guide

**Pull request closed without merging**